plugins/modules/cloudwatch_metric_alarm_info.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: cloudwatch_metric_alarm_info
+version_added: 5.0.0
+short_description: Gather information about the alarms for the specified metric
+description:
+  - Retrieves the alarms for the specified metric.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  alarm_names:
+    description:
+      - The name of the metric.
+    required: false
+    type: list
+    elements: str
+  alarm_name_prefix:
+    description:
+      - An alarm name prefix to retrieve information about alarms that have names that start with this prefix.
+      - Can not be used with alarm_names.
+    required: false
+    type: str
+  alarm_type:
+    description:
+      - Specify this to return metric alarms or composite alarms.
+      - Module is defaulted to return metric alarms but can return composite alarms if I(alarm_type=composite_alarm).
+    required: false
+    type: str
+    default: metric_alarm
+    choices: [composite_alarm, metric_alarm]
+  children_of_alarm_name:
+    description:
+      - If specified returns information about the "children" alarms of the alarm name specified.
+    required: false
+    type: str
+  parents_of_alarm_name:
+    description:
+      - If specified returns information about the "parent" alarms of the alarm name specified.
+    required: false
+    type: str
+  state_value:
+    description:
+      - If specified returns information only about alarms that are currently in the particular state.
+    required: false
+    type: str
+    choices: ['ok', 'alarm', 'insufficient_data']
```

Comment:
I would then provide the choices in uppercase since the API requires uppercase values.